### PR TITLE
Fix wxWidgets to build on an arm processor.

### PR DESCRIPTION
**Pull request by cameronfrandsen**
Define arm neon functions
the three wxWidgets pulls are now ready
https://github.com/smanders/wxcmake/pull/1
https://github.com/smanders/wxWidgets/pull/10

### CODE REVIEW

File: projects/README.md [hunk 1]
```diff
@@ -41,7 +41,7 @@
...
 |[shapelib](http://shapelib.maptools.org/ 'Shapefile C Library website')|[open](http://shapelib.maptools.org/license.html 'MIT Style -or- LGPL')|reading, writing, updating ESRI Shapefiles|1.2.10|[repo](https://github.com/smanders/shapelib 'forked shapelib repo on github')|[diff](https://github.com/smanders/shapelib/compare/v1.2.10...xp1.2.10 'patch/diff')|
 |[wirehair](https://github.com/catid/wirehair 'wirehair repo on github')|[open](https://github.com/catid/wirehair/blob/master/LICENSE.txt 'BSD 3-Clause New or Revised License')|fast and portable fountain codes in C|20.11.21|[repo](https://github.com/catid/wirehair 'wirehair repo on github')|[diff](https://github.com/smanders/wirehair/compare/catid:9e8923a200045c7bdb9e89b449692dbfed6b6de0...xp20.11.21 'patch/diff')|
-|[wxWidgets](http://wxwidgets.org/ 'wxWidgets website')|[open](http://www.wxwidgets.org/about/newlicen.htm 'wxWindows License: essentially LGPL with an exception')|Cross-Platform GUI Library|3.1.0|[repo](https://github.com/wxWidgets/wxWidgets 'wxWidgets repo on github')|[diff](https://github.com/smanders/wxWidgets/compare/v3.1.0_pkg...xp3.1.0 'patch/diff')|
```
Comment (cameronfrandsen) [inline projects/README.md:44]: I figured after https://github.com/smanders/wxcmake/pull/1 and https://github.com/smanders/wxWidgets/pull/10 get merged we could change this back to smanders. Either I can do that or you, its all the same to me.

File: projects/wx.cmake [hunk 2]
```diff
@@ -111,6 +111,14 @@ function(build_wxv)
     if(${CMAKE_C_COMPILER_ID} MATCHES "Clang")
       list(APPEND XP_CONFIGURE_BASE CC=clang)
     endif()
+    execute_process(COMMAND uname --machine
+      OUTPUT_VARIABLE unameMachine
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+      ERROR_VARIABLE unameErr
+      )
+    if(DEFINED unameMachine AND NOT unameErr AND unameMachine MATCHES "^aarch")
+      list(APPEND XP_CONFIGURE_BASE --build=aarch64-unknown-linux-gnu --enable-arm-neon)
```
Comment (cameronfrandsen) [inline projects/wx.cmake:120]: for aarch64 I have to specify the build type and enable the arm neon optimizations. (only effects arm 7 and greater)

File: projects/wx31.cmake [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # wx31
 xpProOption(wx31 DBG_MSVC)
 set(REPO github.com/wxWidgets/wxWidgets)
-set(FORK github.com/smanders/wxWidgets)
+set(FORK github.com/cameronfrandsen/wxWidgets)
```
Comment (smanders) [inline projects/wx31.cmake:4]: for future reference: you shouldn't need to change this... certainly never need to commit it

the only reason I can see committing this is to verify that the README.md patch/diff link shows your commits, but that's really not necessary

you can navigate to the `xpbase/Source/wx31_repo` and add a remote to your fork, set the branch or commit or local changes to what you wish to be in the patch, then go back to the build directory where you ran `XP_STEP=mkpatch` and build -- the patch file in the externpro `patches/` directory should show the changes you want to commit and submit

let me know if you'd like any clarification on this

File: projects/wx.cmake (hunk 2 above)
Comment (smanders) [reply projects/wx.cmake:120]: my plan is to take your changes to this file, and the `patches/wx31.patch` changes and nothing else, for now... unless you convince me that we need the `patches/wxcmake31.patch` changes

but the changes to `projects/README.md`, and `projects/wx31.cmake` and `projects/wxcmake31.cmake` aren't necessary and I'd rather keep the history and blame where it is then merge these changes and then immediately modify these files (and introduce unnecessary history and lose blame on these lines)

**Comment by smanders** [comment]
completed with commits to dev branch referenced above
I did *not* merge the commits in this pull, but the two files with changes necessary for this effort
* patches/wx31.patch
* projects/wx.cmake

were taken care of in the commits referenced above
* wx31.patch: add png/arm source
* wx: aarch build

again @cameronfrandsen please let me know if I've overlooked something in regards to wxcmake31.patch

**Pull request closed without merging**